In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [5]:
df=pd.read_csv("A:\Dataset\car_purchase_data.csv")

<>:1: SyntaxWarning: invalid escape sequence '\D'
<>:1: SyntaxWarning: invalid escape sequence '\D'
C:\Users\astik\AppData\Local\Temp\ipykernel_14924\3958727682.py:1: SyntaxWarning: invalid escape sequence '\D'
  df=pd.read_csv("A:\Dataset\car_purchase_data.csv")


In [7]:
df

Age  EstimatedSalary  Purchased
0     19            19000          0
1     35            20000          0
2     26            43000          0
3     27            57000          0
4     19            76000          0
..   ...              ...        ...
395   46            41000          1
396   51            23000          1
397   50            20000          1
398   36            33000          0
399   49            36000          1

[400 rows x 3 columns]

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 400 entries, 0 to 399
Data columns (total 3 columns):
 #   Column           Non-Null Count  Dtype
---  ------           --------------  -----
 0   Age              400 non-null    int64
 1   EstimatedSalary  400 non-null    int64
 2   Purchased        400 non-null    int64
dtypes: int64(3)
memory usage: 9.5 KB


In [11]:
df['Age'].value_counts()

Age
35    32
37    20
26    16
41    16
42    16
39    15
40    15
48    14
47    14
27    13
38    13
28    12
36    12
46    12
31    11
30    11
49    10
29    10
33     9
32     9
24     9
59     7
19     7
45     7
20     7
60     7
34     6
23     6
52     6
25     6
58     6
53     5
57     5
22     5
18     5
21     4
50     4
54     4
55     3
56     3
51     3
43     3
44     2
Name: count, dtype: int64

In [13]:
df['Purchased'].value_counts()

Purchased
0    257
1    143
Name: count, dtype: int64

It says that 257 people are not purchased and 143 are purchased cars.

In [16]:
age=pd.get_dummies(df['Age'],drop_first=True)
fd=pd.concat([df,age],axis=1)

In [18]:
X=df[['EstimatedSalary','Age']]
y=df['Purchased']

In [20]:
from sklearn.preprocessing import StandardScaler
ss=StandardScaler()
ss.fit(X)

StandardScaler()

In [22]:
sf=ss.transform(X)
sf

array([[-1.49004624, -1.78179743],
       [-1.46068138, -0.25358736],
       [-0.78528968, -1.11320552],
       [-0.37418169, -1.01769239],
       [ 0.18375059, -1.78179743],
       [-0.34481683, -1.01769239],
       [ 0.41866944, -1.01769239],
       [ 2.35674998, -0.54012675],
       [-1.07893824, -1.20871865],
       [-0.13926283, -0.25358736],
       [ 0.30121002, -1.11320552],
       [-0.52100597, -1.11320552],
       [ 0.47739916, -1.6862843 ],
       [-1.51941109, -0.54012675],
       [ 0.35993973, -1.87731056],
       [ 0.30121002, -0.82666613],
       [-1.3138571 ,  0.89257019],
       [-1.28449224,  0.70154394],
       [-1.22576253,  0.79705706],
       [-1.19639767,  0.98808332],
       [-1.40195167,  0.70154394],
       [-0.60910054,  0.89257019],
       [-0.84401939,  0.98808332],
       [-1.40195167,  0.70154394],
       [-1.37258681,  0.79705706],
       [-1.46068138,  0.89257019],
       [-1.22576253,  1.08359645],
       [-1.16703281,  0.89257019],
       [-0.78528968,

In [24]:
df_feat=pd.DataFrame(sf,columns=X.columns)
df_feat

EstimatedSalary       Age
0          -1.490046 -1.781797
1          -1.460681 -0.253587
2          -0.785290 -1.113206
3          -0.374182 -1.017692
4           0.183751 -1.781797
..               ...       ...
395        -0.844019  0.797057
396        -1.372587  1.274623
397        -1.460681  1.179110
398        -1.078938 -0.158074
399        -0.990844  1.083596

[400 rows x 2 columns]

In [26]:
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test=train_test_split(sf,y,test_size=0.3)

In [28]:
from sklearn.neighbors import KNeighborsClassifier
knn=KNeighborsClassifier(n_neighbors=1)
knn.fit(X_train,y_train)

KNeighborsClassifier(n_neighbors=1)

In [30]:
pred=knn.predict(X_test)

In [32]:
pred

array([0, 1, 0, 1, 0, 1, 1, 1, 1, 1, 1, 0, 0, 1, 0, 0, 0, 0, 1, 0, 1, 0,
       0, 1, 1, 1, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 1, 1, 0, 1,
       0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 1, 0, 1, 0,
       0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 1,
       0, 1, 1, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0,
       0, 1, 0, 0, 0, 1, 1, 0, 0, 0], dtype=int64)

In [34]:
from sklearn.metrics import classification_report,confusion_matrix
confusion_matrix(y_test,pred)

array([[64,  8],
       [11, 37]], dtype=int64)

In [36]:
print(classification_report(y_test,pred))

              precision    recall  f1-score   support

           0       0.85      0.89      0.87        72
           1       0.82      0.77      0.80        48

    accuracy                           0.84       120
   macro avg       0.84      0.83      0.83       120
weighted avg       0.84      0.84      0.84       120



This is all about K-Nearest Neighbour and it's classifier.